src/adaptation/AccessCheck.h

```diff
@@ -36,7 +36,7 @@ class AccessCheck: public virtual AsyncJob
 
     // use this to start async ACL checks; returns true if started
     static bool Start(Method method, VectPoint vp, HttpRequest *req,
-                      HttpReply *rep, AccessLogEntry::Pointer &al, Adaptation::Initiator *initiator);
+                      HttpReply *, const AccessLogEntry::Pointer &, Adaptation::Initiator *);
```

Comment:
Done (72352c7).

Reply:
Needs more work to fix compilation [errors](https://semaphoreci.com/squid/squid/branches/pull-request-904/builds/2), including these ones:

```
../../src/stat.cc:1836:46: error: invalid use of incomplete type ‘class AccessLogEntry’
../../src/stat.cc:1837:41: error: invalid use of incomplete type ‘class AccessLogEntry’
../../src/stat.cc:1838:45: error: invalid use of incomplete type ‘class AccessLogEntry’
```

BTW, overall, I think the rule of thumb we should follow when deciding whether to use a forward declaration or a complete one should be something like this:

* Use forward declaration of X for Y if some of Y users (existing or hypothetical, including Y itself if Y is a definition) are correctly compiled with just a forward declaration of X.

Perhaps this can be simplified further:

* Use forward declaration of X for Y if Y is correctly compiled (in an existing or hypothetical context) with just a forward declaration of X.

For example, if a class Y uses a forward declaration for its data member, can Y declaration be compiled? If the answer is no, then class member declarations must use AccessLogEntry::Pointer. Otherwise, they must use AccessLogEntryPointer instead. I do not know what the correct answer is, but I suspect it is "no".